### ISSUE TITLE
Icon Pack lost

### ISSUE THREAD

**Issue by mbruski** [evilbunny2008/weeWxWeatherApp#18]
When I first configured the app I seemed to have everything working the way I wanted but noticed forecast info was in metric.  I had set fctype=weather.gov but didn't toggle the metric switch when first setting up.  Not knowing about the hidden settings menu, I unloaded the app and reloaded from the google store, setup settings URL, toggled the switch like I should have done the first time and also toggled the "icon" switch before saving.  I received an error message about the icon.zip file not being found.  I then performed another app reload and this time left the "icon" switch in the default state.   This time there was no error but the forecast icons are all blank.  How do I get the icons back?

**Comment by evilbunny2008** [comment]
Try deleting the weewx folder from sdcard or internal memory, that should force it to redownload.

**Comment by evilbunny2008** [comment]
After a close look, that won't properly trigger it.

You need to delete the weewx folder and then bring up settings (swiping from left side of the screen), and then make sure icons are  chosen and then click on save settings...

**Comment by mbruski** [comment]
FWIW, I found the weeWX folder in the S10's Internal storage and it actually contains the icon.zip file and the icons folder containing 828 items (all .png images that I can see).  Before I deleted it, I went back to settings and toggled the icon switch, then clicked save.  Here is the complete text of the error I got:

Wasn't able to detect forecast icons.
java.io.FileNotFoundException:/storage/emulated/0/weeWX/icon.zip  open failed:  EACCES (Permission denied)

After deleting the weeWX folder and saving settings I get:

Wasn't able to detect forecast icons.
java.lang.Exception: There was a problem making the icons directory, you will need to try again.

At this point I attempted to reload the app but I continue to get the error about making the icons directory.

**Comment by evilbunny2008** [comment]
It looks like you are having permission problems, have you tried to reboot your phone?

**Comment by mbruski** [comment]
I tried that already but it didn't help.  I was able to recreate the weeWX folder but it now sits there empty.  Is there someplace I can download the icons.zip file from to preload the folder?Sent from Samsung Galaxy smartphone.
-------- Original message --------From: evilbunny2008 <notifications@github.com> Date: 9/19/20  07:48  (GMT-05:00) To: evilbunny2008/weeWXWeatherApp <weeWXWeatherApp@noreply.github.com> Cc: Mike <aj9x@comcast.net>, Author <author@noreply.github.com> Subject: Re: [evilbunny2008/weeWXWeatherApp] Icon Pack lost (#18) 
It looks like you are having permission problems, have you tried to reboot your phone?

**Comment by mbruski** [comment]
I think I've tried every combo of delete folder/file/app I can think of and still can't shake these access errors.  I even tried manually putting a copy of the zip file in the weeWX folder but to no avail.I was trying to decipher what happens to app files/folders in /storage/emulated/0/ and what happens to various types of files when the app is deleted but am totally befuddled now.  I dont know what is causing the access issue because I gave the app permission to access files on first running it.  Nothing else on the phone has changed since I first saw the icons displayed.  For now it looks like I'll have to abandon using weather.gov.  I'll probably give fctype=yahoo a whirl tomorrow and see how that goes.Sent from Samsung Galaxy smartphone.
-------- Original message --------From: aj9x <aj9x@comcast.net> Date: 9/19/20  08:21  (GMT-05:00) To: evilbunny2008/weeWXWeatherApp <reply@reply.github.com>, evilbunny2008/weeWXWeatherApp <weeWXWeatherApp@noreply.github.com> Cc: Author <author@noreply.github.com> Subject: Re: [evilbunny2008/weeWXWeatherApp] Icon Pack lost (#18) 
I tried that already but it didn't help.  I was able to recreate the weeWX folder but it now sits there empty.  Is there someplace I can download the icons.zip file from to preload the folder?Sent from Samsung Galaxy smartphone.-------- Original message --------From: evilbunny2008 <notifications@github.com> Date: 9/19/20  07:48  (GMT-05:00) To: evilbunny2008/weeWXWeatherApp <weeWXWeatherApp@noreply.github.com> Cc: Mike <aj9x@comcast.net>, Author <author@noreply.github.com> Subject: Re: [evilbunny2008/weeWXWeatherApp] Icon Pack lost (#18) 
It looks like you are having permission problems, have you tried to reboot your phone?

**Comment by evilbunny2008** [comment]
The only thing I can think of is somehow it didn't get permission to read/write to internal memory when prompted...

TL;DR Uninstall the app, then reinstalling it and giving it storage permissions.

The slightly longer version is to check for storage permission. This will vary by phone, but for me I went into the settings app on the phone, then Apps & notifications, then Storage & memory, then See all apps, then select weeWX, then Permissions.

Make sure it has access to storage.

**Comment by BotsWithLasers** [comment]
So while OP didn't mention which android version they where running I have found that  the app is broken on Android 11 as I am getting that exact problem with icons and the app generally doesn't work.     The new Scoped permission security in Android 11 defaults to on for all apps even if they are targeted at a lower SDK.  Which seems to be breaking the apps ability to create/access a /weeWX folder in internal storage.

I had to modify and compile a version to get it work on Android 11.

According to settings in project in the repo the app's  in default config is using Target SDK:  API 28, Min SDK: 19 and Compile SDk is set to 28.

From what I found it needs Compile SDK changed to 29,  and needs:

android:requestLegacyExternalStorage="true"

added to application section of the AndroidManifest.xml file.    That seems to get it to compile and work on Android 11 as SDK 29 (android 10) has ability to have applications ask to use the old security model with the requestLegacyExternalStorage attribute.

Making those two changes, was able to build a release version and install it on my Samsung N20 with Android 11 and everything seems to be working fine.

Normally would do a pull request with the changes, but with the holidays I don't have the time at the moment.

**Comment by evilbunny2008** [comment]
Thanks for that, I made the changes and uploaded a new release to the play store, it just needs approving now.

UPDATE: app is now live

**Comment by BotsWithLasers** [comment]
I uninstall my compiled version, blew away the weeWX folder, and reinstalled from the play store.   Ran it and looks like everything is working great.   It successfully created the new weeWX folder, downloaded the icons.zip file and create the icons directory.    Did that a process a couple times and everything worked every time.   

So everything seems to be working without any issues on Android 11.   Hopefully this fixes the issue for the OP as well.  But at least for the Android 11 users I think this solves the permissions issues with the weeWX folder.    

Not sure if this workaround for permissions will carry on going forward on android versions.  As not sure how long it will be before they force all apps to use the actual Scoped Storage Api's and stop honoring the Legacy flag.     Which means eventually app will need to be updated to support the new Scoped Storage API's.

**Comment by mbruski** [comment]
I'm on a S10 with Android 10.   I installed the latest version from the app store a few minutes ago and everything is working fine now.   Looks like the problem is resolved.   Thanks to both of you for sorting that out.   Mike - AJ9X
